zebra-state/src/service/finalized_state/disk_db.rs

```diff
@@ -321,34 +322,20 @@ impl ReadDisk for DiskDb {
     fn zs_first_key_value<C, K, V>(&self, cf: &C) -> Option<(K, V)>
     where
         C: rocksdb::AsColumnFamilyRef,
-        K: FromDisk,
+        K: IntoDisk + FromDisk,
         V: FromDisk,
     {
-        // Reading individual values from iterators does not seem to cause database hangs.
-        self.db
-            .iterator_cf(cf, rocksdb::IteratorMode::Start)
-            .next()?
-            .map(|(key_bytes, value_bytes)| {
-                Some((K::from_bytes(key_bytes), V::from_bytes(value_bytes)))
-            })
-            .expect("unexpected database failure")
+        self.zs_seek(cf, |iter| iter.seek_to_first())
```

Comment:
This is a substantial change which could change performance. Can you do a full sync before we merge this PR? Or split this change into a separate PR?

Reply:
Does each of the methods that calls zs_seek have enough test coverage to make sure it works correctly? Or do we need to add a Some and None test for the methods that aren't called from production code?

Reply:
Split out to #7759.